server.py

```diff
@@ -1,58 +1,78 @@
-import threading
-from flask import Flask
-from flask_socketio import SocketIO, emit
-from canvas_board import CanvasBoard, CanvasBoardEncoder
+import random, string
+from flask import Flask, jsonify
+from flask_socketio import SocketIO, emit, join_room, leave_room
+from canvas_board import CanvasBoard
+from json import dumps
 
 # constants
 PORT = 8080
 HEIGHT = 500
 WIDTH = 500
+ID_LENGTH = 7
 
 # create Flask object
-app = Flask(__name__)
+app = Flask(__name__, template_folder='static/', static_folder='static/')
 app.config['SECRET_KEY'] = 'secret!'
 socketio = SocketIO(app, cors_allowed_origins='*')
 
 # server class
 class Server:
-    def __init__(self, imagedata):
-        self.board = CanvasBoard(imagedata)
-        self.lock = threading.Lock()
-
-    def update_board(self, diffs: list):
-        self.lock.acquire()
-        self.board.update_board(diffs)
-        self.lock.release()
+    def __init__(self, boards: 'dict that maps room id to CanvasBoard'):
+        self.boards = boards
 
+    def update_board(self, diffs: list, room_id: str):
+        self.boards[room_id].update_board(diffs)
 
 # instantiate server class for board state
 imagedata = {
     'width': WIDTH,
     'height': HEIGHT,
     'data': [0 for i in range(4 * WIDTH * HEIGHT)]
 }
-server = Server(imagedata)
+server = Server({})
 
 @app.route('/')
 def index():
     return app.send_static_file('index.html')
 
+@app.route('/create')
+def create():
+    room = None
+    while room in server.boards or room is None:
+        room = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(ID_LENGTH))
+        print(room)

```

Comment:
Consider using uuid4(): https://docs.python.org/3/library/uuid.html#uuid.uuid4

Reply:
Alternatively, the endpoint can receive a room ID as a part of its request and we can use that to create a new room (or not if one already exists).  Then we will just return success or fail with helpful error message.

We will probably want to mark this as a post as well.